packedForest/src/forestTypes/unsupervisedForests/urf/unprocessedURFNode.h

```diff
@@ -0,0 +1,183 @@
+							fpSingleton::getSingleton().prefetchFeatureVal(featuresToTry.back(),baseUnprocessedNodeUnsupervised<T>::obsIndices->returnInSample(q));
+						}
+
+						for(int i =0; i < baseUnprocessedNodeUnsupervised<T>::obsIndices->returnInSampleSize(); ++i){
+							baseUnprocessedNodeUnsupervised<T>::featureHolder[i] = fpSingleton::getSingleton().returnFeatureVal(featuresToTry.back(), baseUnprocessedNodeUnsupervised<T>::obsIndices->returnInSample(i));
+						}
+					}
+				}
+
+				inline void setupNode(){
+					pickMTRY();
+					baseUnprocessedNodeUnsupervised<T>::setHolderSizes();
+					baseUnprocessedNodeUnsupervised<T>::setNodeImpurity(baseUnprocessedNodeUnsupervised<T>::calculateNodeImpurity());
+				}
+
+
+				inline bool goLeft(const int index){
+					T featureVal = fpSingleton::getSingleton().returnFeatureVal(bestSplitInfo.returnFeatureNum(),index);
+
+					if(featureVal <= bestSplitInfo.returnSplitValue()){
+						return true;
+					}else{
+						return false;
+					}
+				}
+
+				inline void deleteObsIndices(){
+					delete baseUnprocessedNodeUnsupervised<T>::obsIndices;
+					baseUnprocessedNodeUnsupervised<T>::obsIndices = NULL;
+				}
+
+				inline void moveDataLeftOrRight(){
+
+					baseUnprocessedNodeUnsupervised<T>::leftIndices = new stratifiedInNodeClassIndicesUnsupervised();
+					baseUnprocessedNodeUnsupervised<T>::rightIndices = new stratifiedInNodeClassIndicesUnsupervised();
+
+					int lNum =0;
+					int rNum =0;
+					for (int i=0; i < baseUnprocessedNodeUnsupervised<T>::obsIndices->returnInSampleSize();++i){
+						if(goLeft(baseUnprocessedNodeUnsupervised<T>::obsIndices->returnInSample(i))){
+							++lNum;
+							baseUnprocessedNodeUnsupervised<T>::leftIndices->addIndexToInSamples(baseUnprocessedNodeUnsupervised<T>::obsIndices->returnInSample(i));	
+						}else{
+							++rNum;
+							baseUnprocessedNodeUnsupervised<T>::rightIndices->addIndexToInSamples(baseUnprocessedNodeUnsupervised<T>::obsIndices->returnInSample(i));	
+						}
+					}
+                                        if(lNum <= 0 || rNum <= 0){
+                                                lNum++;
+                                                rNum--;
+                                        }
+
+					assert(lNum > 0);
+					assert(rNum > 0);
+
+
+					for (int i=0; i < baseUnprocessedNodeUnsupervised<T>::obsIndices->returnOutSampleSize();++i){
+						if(goLeft(baseUnprocessedNodeUnsupervised<T>::obsIndices->returnOutSample(i))){
+							baseUnprocessedNodeUnsupervised<T>::leftIndices->addIndexToOutSamples(baseUnprocessedNodeUnsupervised<T>::obsIndices->returnOutSample(i));	
+						}else{
+							baseUnprocessedNodeUnsupervised<T>::rightIndices->addIndexToOutSamples(baseUnprocessedNodeUnsupervised<T>::obsIndices->returnOutSample(i));	
+						}
+					}
+					deleteObsIndices();
+				}
+
+
+				inline void findBestSplit(){
+					//timeLogger logTime;
+					splitURF<T> findSplit; //This is done twice
+					//TODO This needs to change to real mtry
```

Comment:
Should this TOTO be made into an issue? @falkben

Reply:
@MrAE @falkben I addressed most of the comments. Let me know if I missed something.